TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/RoutesToProfessionalStatus/EditRoute/CheckYourAnswers.cshtml.cs

```diff
@@ -154,23 +154,40 @@ await fileService.GetFileUrlAsync(ChangeReasonDetail.EvidenceFileId!.Value, File
+            || JourneyInstance!.State.TrainingStartDate is not null) == false ? NotCompleteError.StartDate : NotCompleteError.None;
+        errors = (QuestionDriverHelper.FieldRequired(route.TrainingEndDateRequired, status.GetEndDateRequirement()) != FieldRequirement.Mandatory
+            || JourneyInstance!.State.TrainingEndDate is not null) == false ? errors | NotCompleteError.EndDate : errors;
+        errors = (QuestionDriverHelper.FieldRequired(route.AwardDateRequired, status.GetAwardDateRequirement()) != FieldRequirement.Mandatory
+            || JourneyInstance!.State.AwardedDate is not null) == false ? errors | NotCompleteError.AwardDate : errors;
+        errors = (QuestionDriverHelper.FieldRequired(route.DegreeTypeRequired, status.GetDegreeTypeRequirement()) != FieldRequirement.Mandatory
+            || JourneyInstance!.State.DegreeTypeId is not null) == false ? errors | NotCompleteError.DegreeType : errors;
+        errors = (QuestionDriverHelper.FieldRequired(route.TrainingAgeSpecialismTypeRequired, status.GetAgeSpecialismRequirement()) != FieldRequirement.Mandatory
+            || (JourneyInstance!.State.TrainingAgeSpecialismRangeFrom is not null && JourneyInstance!.State.TrainingAgeSpecialismRangeTo is not null)
+            || JourneyInstance!.State.TrainingAgeSpecialismType is not null) == false ? errors | NotCompleteError.AgeSpecialism : errors;
+        errors = (QuestionDriverHelper.FieldRequired(route.TrainingCountryRequired, status.GetCountryRequirement()) != FieldRequirement.Mandatory
+            || JourneyInstance!.State.TrainingCountryId is not null) == false ? errors | NotCompleteError.Country : errors;
+        errors = (QuestionDriverHelper.FieldRequired(route.TrainingSubjectsRequired, status.GetSubjectsRequirement()) != FieldRequirement.Mandatory
+            || !JourneyInstance!.State.TrainingSubjectIds.IsNullOrEmpty()) == false ? errors | NotCompleteError.Subject : errors;
+        errors = (QuestionDriverHelper.FieldRequired(route.TrainingProviderRequired, status.GetTrainingProviderRequirement()) != FieldRequirement.Mandatory
+            || JourneyInstance!.State.TrainingProviderId is not null) == false ? errors | NotCompleteError.Provider : errors;
+        errors = (QuestionDriverHelper.FieldRequired(route.InductionExemptionRequired, status.GetInductionExemptionRequirement()) != FieldRequirement.Mandatory
+            || JourneyInstance!.State.IsExemptFromInduction is not null) == false ? errors | NotCompleteError.InductionExemption : errors;
+
+        return errors == NotCompleteError.None;
+    }
+
+    enum NotCompleteError
+    {
+        None = 0,
+        StartDate = 1,
+        EndDate = 1 << 1,
+        AwardDate = 1 << 2,
+        DegreeType = 1 << 3,
+        AgeSpecialism = 1 << 4,
+        Country = 1 << 5,
+        Subject = 1 << 6,
+        Provider = 1 << 7,
+        InductionExemption = 1 << 8
```

Comment:
The logic above is the same but I've started collecting the errors, partly so I could see what was going on, and also because at some point the FE is going to have to do something to help the user rectify the errors.